nzgd_map/views.py

```diff
@@ -23,41 +23,186 @@
+                    uploaded_df["longitude"] <= 180
+                )
+                uploaded_df = uploaded_df[valid_lat & valid_lon]
+                if uploaded_df.empty:
+                    flask.flash(
+                        "Uploaded data from session contained no valid geolocations after range filtering.",
+                        "warning",
+                    )
+
+            if (
+                uploaded_df.empty
+            ):  # If all rows became NaN and were dropped or filtered out
+                uploaded_df = None  # Ensure uploaded_df is None if it becomes empty
+                if (
+                    "uploaded_locations" in flask.session
+                ):  # Clean up session if df becomes empty
+                    del flask.session["uploaded_locations"]
+                # flask.flash("Uploaded data from session was invalid or empty after type conversion.", "warning") # Original message, can be removed or kept
+            else:
+                flask.flash(
+                    f"Loaded {len(uploaded_df)} valid locations from session.", "info"
+                )
+                # Re-create hovertext if not present, as it might not survive JSON serialization well depending on content
+                if (
+                    "hovertext" not in uploaded_df.columns and not uploaded_df.empty
+                ):  # Check not empty again
+                    uploaded_df["hovertext"] = "Uploaded: " + uploaded_df.index.astype(
+                        str
+                    )
+        except Exception as e:
+            uploaded_df = None  # Error during deserialization
+            if "uploaded_locations" in flask.session:  # Check before deleting
+                del flask.session["uploaded_locations"]  # Clear corrupted data
+            flask.flash(
+                f"Error loading uploaded locations from session: {e}. Please re-upload if needed.",
+                "error",
+            )
+    # --- End: Load uploaded_df from session ---
+
+    if flask.request.method == "POST":
+        vs30_correlation = flask.request.form.get(
+            "vs30_correlation", default=constants.default_vs_to_vs30_correlation
+        )
+        spt_vs_correlation = flask.request.form.get(
+            "spt_vs_correlation", default=constants.default_spt_to_vs_correlation
+        )
+        cpt_vs_correlation = flask.request.form.get(
+            "cpt_vs_correlation", default=constants.default_cpt_to_vs_correlation
+        )
+        colour_by = flask.request.form.get("colour_by", default="vs30")
+        hist_by = flask.request.form.get("hist_by", default="vs30_log_residual")
+        query = flask.request.form.get("query")
+
+        # --- Start: Handle CSV file upload ---
+        if "csv_file" in flask.request.files:
+            file = flask.request.files["csv_file"]
+            if file.filename != "":  # A file was actually selected for upload
+                if file.filename.endswith(".csv"):
```

Comment:
similarly to above this can be simplified significantly:

```suggestion
        # --- Start: Handle CSV file upload ---
        file = flask.request.files.get('csv_files')
        if file and file.filename.endswith('.csv'):
```

Reply:
Note the non-empty filename check is unnecessary because `''` does not end with `'.csv'`